code/procs/jobprocs.dm

```diff
@@ -50,6 +50,30 @@
 
 	return candidates
 
+/proc/CanPlayerBeAssignedJob(var/mob/new_player/player, var/datum/job/JOB)
+	if (ticker?.mode && istype(ticker.mode, /datum/game_mode/revolution))
+		if(checktraitor(player) && (JOB.cant_spawn_as_rev || JOB.cant_spawn_as_con))
+			// Fixed AI, security etc spawning as rev heads. The special job picker doesn't care about that var yet,
+			// but I'm not gonna waste too much time tending to a basically abandoned game mode (Convair880).
+			return FALSE
+	if (!JOB || jobban_isbanned(player,JOB.name))
+		return FALSE
+	if (JOB.needs_college && !player.has_medal("Unlike the director, I went to college"))
```

Comment:
wait what lol

Reply:
lol yeah, I think I just copied this functionality from the loop that was assigning favourite jobs. I just looked again and there are variations of this logic in a couple of places (like in the `FindOccupationCandidates` function).

I can have a look at pulling these into a standardised function too.